### DISCUSSION TITLE
Fuzzy search for easier model search

### DISCUSSION THREAD

**Discussion by SirVeggie** [civitai/civitai#519] [Ideas]
Currently the search feature is honestly not great, even the search box autocomplete feature finds models easier than pressing enter to do a search. Please implement approximate matching of search strings so you don't need to know the exact model name to find it.

Example: model named "Diffusion Brush ( Everything )" can be found by the search term "diffusion brush", but searches "diffusionbrush" or "diffusion everything" find nothing.

**Comment by Ristellise** [comment]
Gonna leave a suggestion here since it's closely related. But you could bring in a *Full Text Search engine* for the search bar (Elasticsearch/OpenSearch, Meilisearch).  

I'm noticing it is getting more difficult getting the right hits on model names, and I've read of issues relating to search (e.g. maybe `headpat lora` was not caught with the search `head pat` or some other typo issues.)

Personally I've toyed around with Meilisearch after finding ES/OS being too much a hassle to setup (and being very biased towards it since I haven't tried others and it's quite fast). From python it was rather simple to implement, and they also provide bindings for [TypeScript/JS](https://github.com/meilisearch/meilisearch-js). Their tokenization as of late support CJK content as well.

EDIT: Apparenlty been told by a chirpy bird that Postgress also has Full Text Search. Maybe consider using that?

**Comment by PeterNjeim** [comment]
Edit: they've implemented an extremely good search system now, even works with filenames. Original comment below:

Another issue is that I see a LoRA name in a prompt and try to search for it but it yields no results. The problem is that the title of the model is different than the file name. Adding filenames to the search index, as well as other metadata such as the description, version numbers, model hashes (already supported), and others should improve the search
